UserTools/VtxSeedFineGrid/VtxSeedFineGrid.cpp

```diff
@@ -0,0 +1,470 @@
+
+
+bool VtxSeedFineGrid::Initialise(std::string configfile, DataModel &data){
+
+  /////////////////// Useful header ///////////////////////
+  if(configfile!="") m_variables.Initialise(configfile); // loading config file
+  //m_variables.Print();
+  m_variables.Get("verbosity", verbosity);
+  m_variables.Get("useTrueDir", useTrueDir);
+  m_variables.Get("useSimpleDir", useSimpleDir);
+  m_variables.Get("useMRDTrack", useMRDTrack);
+  m_variables.Get("usePastResolution", usePastResolution);
+  m_variables.Get("useDirectionGrid", useDirectionGrid);
+  m_variables.Get("multiGrid", multiGrid);
+  m_variables.Get("InputFile", InputFile);
+
+
+  m_data= &data; //assigning transient data pointer
+  /////////////////////////////////////////////////////////////////
+  vSeedVtxList = nullptr;
+
+  return true;
+}
+
+
+bool VtxSeedFineGrid::Execute(){
+	Log("VtxSeedFineGrid Tool: Executing", v_debug, verbosity);
+
+	auto* annie_event = m_data->Stores["RecoEvent"];
+	if (!annie_event) {
+		Log("Error: VtxSeedFineGrid tool could not find the RecoEvent Store", v_error, verbosity);
+		return false;
+	}
+
+	m_data->Stores.at("RecoEvent")->Get("vSeedVtxList", vSeedVtxList);
+
+	auto get_vtx = m_data->Stores.at("RecoEvent")->Get("TrueVertex", fTrueVertex);  ///> Get digits from "RecoEvent" 
+/*	if (!get_vtx) {
+		Log("LikelihoodFitterCheck  Tool: Error retrieving TrueVertex! ", v_error, verbosity);
+		return false;
+	}*/
+
+	// check if event passes the cut
+	bool EventCutstatus = false;
+	auto get_evtstatus = m_data->Stores.at("RecoEvent")->Get("EventCutStatus", EventCutstatus);
+	if (!get_evtstatus) {
+		Log("Error: The PhaseITreeMaker tool could not find the Event selection status", v_error, verbosity);
+		//return false;
+                Log("running with reco cuts or none.",v_debug,verbosity);
+                EventCutstatus = true;
+	}
+
+	if (!EventCutstatus) {
+		Log("Message: This event doesn't pass the event selection. ", v_message, verbosity);
+		return true;
+	}
+
+	// Retrive digits from RecoEvent
+	auto get_digit = m_data->Stores.at("RecoEvent")->Get("RecoDigit", fDigitList);  ///> Get digits from "RecoEvent" 
+	if (!get_digit) {
+		Log("LikelihoodFitterCheck  Tool: Error retrieving RecoDigits,no digit from the RecoEvent!", v_error, verbosity);
+		return false;
+	}
+	auto get_flagsapp = m_data->Stores.at("RecoEvent")->Get("EventFlagApplied",fEventStatusApplied);
+    auto get_flags = m_data->Stores.at("RecoEvent")->Get("EventFlagged",fEventStatusFlagged); 
+    if(!get_flagsapp || !get_flags) {
+      Log("PhaseITreeMaker tool: No Event status applied or flagged bitmask!!", v_error, verbosity);
+      //return false;	
+    }
+    // check if event passes the cut
+    if((fEventStatusFlagged) != 0) {
+      Log("PhaseIITreeMaker Tool: Event was flagged with one of the active cuts.",v_debug, verbosity);
+      return true;	
+    }
+
+	if (useSimpleDir) {
+		Log("Using simple direction", v_debug, verbosity);

```

Comment:
do these checks/printouts need to be in Execute? It seems like they only need to run once in Initialise.